fml/status.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_STATUS_H_
+#define FLUTTER_FML_STATUS_H_
+
+#include "flutter/fml/string_view.h"
+
+namespace fml {
+
+/// Google's canonical list of errors, should be in sync with:
+/// https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto
+enum class StatusCode : int {
+  kOk = 0,
+  kCancelled = 1,
+  kUnknown = 2,
+  kInvalidArgument = 3,
+  kDeadlineExceeded = 4,
+  kNotFound = 5,
+  kAlreadyExists = 6,
+  kPermissionDenied = 7,
+  kResourceExhausted = 8,
+  kFailedPrecondition = 9,
+  kAborted = 10,
+  kOutOfRange = 11,
+  kUnimplemented = 12,
+  kInternal = 13,
+  kUnavailable = 14,
+  kDataLoss = 15,
+  kUnauthenticated = 16,
+  kDoNotUseReservedForFutureExpansionUseDefaultInSwitchInstead_ = 20
```

Comment:
This does not apply to us and none of them require values.

Reply:
There is no reason to go out of our way to be incompatible with absl.  These values are just as good as anything else, no?

Reply:
I'd rather not have FML become an agglomeration of things from absl. This was already how some stuff from Base and FXL got in this library with (in some cases) limited use. Let's build and use only the things we need.

Reply:
K, done.